### PR TITLE
Vive leap

### PR DESCRIPTION
**Pull request by stakach**
new vive leap protocol driver

### CODE REVIEW

File: drivers/lutron/vive_leap.cr [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+    logger.debug { "Lutron sent: #{data}" }
+    request = Request.from_json(data)
+
+    url = request["Url"]?
+    status = request["StatusCode"]? || "200 OK"
+    message_type = request["MessageBodyType"]?
+
+    # process the message based on its type by preference
+    case message_type
+    when "OneClientSettingDefinition"
```
Comment (tassja) [inline drivers/lutron/vive_leap.cr:153]: Maybe make an enum for the MessageBodyTypes so you can do `case in...` to ensure exhaustive coverage?

**Review by tassja — DISMISSED** [review]

File: drivers/lutron/vive_leap.cr (hunk 1 above)
Comment (tassja) [reply drivers/lutron/vive_leap.cr:153]: Style spec is also failing

**Review by tassja — COMMENTED** [review]
Style spec is also failing

File: drivers/lutron/vive_leap.cr (hunk 1 above)
Comment (stakach) [reply drivers/lutron/vive_leap.cr:153]: I personally don't think it's required here as we only care about a small subset of the message body types

**Pull request merged**